tycho-plugins/repository-utils/src/main/java/org/jboss/tools/tycho/sitegenerator/GenerateRepositoryFacadeMojo.java

```diff
@@ -378,6 +378,10 @@ private void generateWebStuff(File outputRepository, File outputSiteXml) throws
 		} catch (Exception ex) {
 			throw new MojoExecutionException("Error writing file " + indexName, ex);
 		}
+		
+		// rename site.xml to category.xml as site.xml is no longer needed (and can prevent p2 for correctly resolving metadata)
+		// or should we just delete this file entirely?
+		outputSiteXml.renameTo(new File(outputSiteXml.toString().replaceAll("site.xml","category.xml")));
```

Comment:
What about changing it directly in line 439, so we directly create a category.xml instead of a site.xml ?